### PR TITLE
Optimize tryparse

### PR DESCRIPTION
**Pull request by Bond-009**
* Don't check for null before
* Don't try different formats when not needed (`NumberStyles.Integer` is the fast path)

### CODE REVIEW

**Review by Shadowghost — APPROVED** [review]

**Review by crobibero — APPROVED** [review]

File: Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs [hunk 1]
```diff
@@ -168,28 +168,24 @@ private string GetChannelNumber(string extInf, Dictionary<string, string> attrib
...
                 if (attributes.TryGetValue("tvg-id", out attributeValue))
                 {
-                    if (double.TryParse(attributeValue, NumberStyles.Any, CultureInfo.InvariantCulture, out _))
+                    if (double.TryParse(attributeValue, CultureInfo.InvariantCulture, out _))
                     {
                         numberString = attributeValue;
                     }
-                    else if (attributes.TryGetValue("channel-id", out attributeValue))
+                    else if (attributes.TryGetValue("channel-id", out attributeValue)
+                        && double.TryParse(attributeValue, CultureInfo.InvariantCulture, out _))
```
Comment (cvium) [inline Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs:186]: identation

File: Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs [hunk 2]
```diff
@@ -255,19 +251,14 @@ private string GetChannelNumber(string extInf, Dictionary<string, string> attrib
...
-            if (string.IsNullOrWhiteSpace(numberString) ||
-                string.Equals(numberString, "-1", StringComparison.OrdinalIgnoreCase) ||
-                string.Equals(numberString, "0", StringComparison.OrdinalIgnoreCase))
-            {
-                return false;
-            }
-
-            if (!double.TryParse(numberString, NumberStyles.Any, CultureInfo.InvariantCulture, out _))
+            if (string.IsNullOrWhiteSpace(numberString)
+                || string.Equals(numberString, "-1", StringComparison.OrdinalIgnoreCase)
```
Comment (cvium) [inline Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs:255]: I know this is just a formatting change, but IgnoreCase for -1 and 0 doesn't seem necessary

File: Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs:186]: What's wrong with it?

Comment (cvium) [reply Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs:186]: I thought it was supposed to be indented like this?:

```
    attributes.TryGetValue("channel-id", out attributeValue)
    && double.TryParse(attributeValue, CultureInfo.InvariantCulture, out _)
```

**Review by cvium — APPROVED** [review]

File: Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs (hunk 1 above)
Comment (Bond-009) [reply Emby.Server.Implementations/LiveTv/TunerHosts/M3uParser.cs:186]: Not afaik, I've always just indented one level, it just happens to line up with `if (`
Looking trough the code now I did find multiple places where you did this :p

**Pull request merged**